spring-batch-infrastructure/src/test/java/org/springframework/batch/support/transaction/TransactionAwareBufferedWriterTests.java

```diff
@@ -337,6 +338,27 @@ public void testResourceKeyCollision() throws Exception {
 		}				
 	}
 
+	//BATCH-3745
+	@Test
+	public void testWriteInTransactionWithOffset() throws IOException{
+		ArgumentCaptor<ByteBuffer> bb = ArgumentCaptor.forClass(ByteBuffer.class);
+		when(fileChannel.write(bb.capture())).thenReturn(1);
+
+		new TransactionTemplate(transactionManager).execute((TransactionCallback<Void>) status -> {
+
+			try {
+				writer.write("foo", 2, 1);
```

Comment:
The problem I see is that the following write statement also passes the assertion:
 
`writer.write("foo", 1, 1);`

I would propose to use a longer string in which case we validate a sequence of characters are written to the ByteBuffer?

Reply:
Let's indeed use a longer string that illustrates the issue; might I propose extracting `urge` from `hamburger`? It's another somewhat common example based on the `String.substring(int,int)` JavaDocs.

Reply:
Extracting urge from hamburger is not the best choice because this corresponds to off = 4 and len = 4. The bug is reported for off > len. But it makes sense to use a longer string.